File: src/py_package/py_package/robot_news_station.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class RobotNewsStation(Node):
    def __init__(self):
        super().__init__("robot_news_station")
        self.name_ = "C3PO"
        self.publishers_ = self.create_publisher(String, "robot_news", 10)
        self.timer_ = self.create_timer(1,self.publish_news)
        self.contor_ = 0

        self.get_logger().info("Hello. News station robot C3PO is ready! ")


    def publish_news(self):
        msg = String()
        msg.data = f"Hello this is {self.name_} from robot news station with message: {self.contor_}"
        self.publishers_.publish(msg=msg)
        self.contor_ += 1


def main(args=None):
    rclpy.init(args=args)

    node = RobotNewsStation()

    rclpy.spin(node)

    rclpy.shutdown()


if __name__ == "__main__":
    main()        

File: src/cpp_package/src/robot_news_station.cpp
#include "cpp_package/robot_news_station.hpp"

RobotNewsStation::RobotNewsStation() : Node("robot_news_station"), counter_(0)
{
    publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
    timer_ = this->create_wall_timer(std::chrono::seconds(1), std::bind(&RobotNewsStation::PublishNews, this));
    RCLCPP_INFO(this->get_logger(), "Robot News station");
}

void RobotNewsStation::PublishNews()
{
   auto msg = example_interfaces::msg::String();
   msg.data = std::string("Hi, this is R2D2 from the robot news station with message: ") + std::to_string(counter_);
   publisher_->publish(msg);
   counter_++;
}

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<RobotNewsStation>();

    rclcpp::spin(node);

    rclcpp::shutdown();
    return 0;
}